release 2.9.0 blog post (#449)
MDAnalysis/MDAnalysis.github.io: _posts/2025-03-11-release-2.9.0.md

---
layout: post
title: Release 2.9.0 of MDAnalysis
---

We are happy to release version 2.9.0 of MDAnalysis!

This is a minor release of the MDAnalysis library, which means that it
contains enhancements, bug fixes, deprecations, and other
backwards-compatible changes.

## Supported environments

This release supports NumPy 2.0+ and offers backwards compatibility
through to NumPy 1.23.2.

Supported Python versions: **3.10, 3.11, 3.12, 3.13**.

Supported Operating Systems:
  - **Linux** (on [x86-64][])
  - **Windows** (on [x86-64][])
  - **MacOS** (on [x86-64][] and [ARM64][])

## Upgrading to MDAnalysis version 2.9.0

To update with `mamba` (or `conda`)  from the [conda-forge channel][] run

```bash
mamba update -c conda-forge mdanalysis
```

To update from [PyPi][] with `pip` run

```bash
python -m pip install --upgrade MDAnalysis
```

For more help with installation see the [installation instructions in the User Guide][]. 
Make sure you are using a Python version compatible with MDAnalysis 
before upgrading (Python >= 3.10).

## Notable changes

> For a full list of changes, bugfixes and deprecations see the [CHANGELOG][].

Reviewer: ```suggestion
For a full list of changes, bug fixes and deprecations see the [CHANGELOG][].
```

---
layout: post
title: Release 2.9.0 of MDAnalysis
---

We are happy to release version 2.9.0 of MDAnalysis!

This is a minor release of the MDAnalysis library, which means that it
contains enhancements, bug fixes, deprecations, and other

> backwards-compatible changes.

Reviewer 2: Is this actually true for this one? Should we mention https://github.com/MDAnalysis/mdanalysis/issues/4899?

Reviewer 2: I think it's fine, thanks.

## Supported environments

This release supports NumPy 2.0+ and offers backwards compatibility
through to NumPy 1.23.2.

Supported Python versions: **3.10, 3.11, 3.12, 3.13**.

> Please note that Python 3.13 is limited to [PyPi][] for now, the [conda-forge channel][] installs only provide support for Python 3.10 to 3.12.

Author: @hmacdope @IAlibay at the moment there are no py3.13 c-f packages https://anaconda.org/conda-forge/mdanalysis/files?page=1 so we have to keep this line (see https://github.com/MDAnalysis/mdanalysis/issues/4805 ).

Is there a chance that this gets fixed soon (within a day) or should we publish the post as is?

Reviewer 3: I'm travelling sorry, it's a simple migration if others have time though.

Author: @hmacdope do you know what to do?

Author: Thank you both, we now have 3.13 packages.

Make sure you are using a Python version compatible with MDAnalysis 
before upgrading (Python >= 3.10).

## Notable changes

For a full list of changes, bugfixes and deprecations see the [CHANGELOG][].

#### Enhancements:

* Support for Gromacs v2024.4 and v2025 TPR files.
* A new "water" keyword for selecting water residues.
* Support for [distopia][] 0.4.0 as an optional distance library backend.
* Parallel analysis support for `MDAnalysis.analysis.nucleicacids`, `MDAnalysis.analysis.contacts`, and `MDAnalysis.analysis.density`.
* XYZ writer can now write positions up to a user-supplied precision via the `precision` keyword argument.
  

#### Changes:

* `MDAnalysis.analysis.hole2`, `MDAnalysis.analysis.psa`, and
  `MDAnalysis.analysis.waterdynamics`, are now only available via
  optional dependencies (the mdahole2, pathsimanalysis, and
  waterdynamics [MDAKits][]). By default these will not be installed, you
  will need to install them separately if you wish to use them from
  within `MDAnalysis.mdanalysis`. Please
  note that in v3.0 theses modules will be removed completely from the
  MDAnalysis core library and you will need to use the MDAKits
  directly.

> * `fasteners` dependency has been switch to `filelock`.

Reviewer 3: ```suggestion
* `fasteners` dependency has been switched to `filelock`.
```

* Codebase is now formatted with black (version `24`)

## Author statistics

This release was the work of 10 contributors, **3** of which are **new contributors**.

Our **new contributors** are:

* @ChiahsinChu
* @tanishy7777
* @jauy123

## Acknowledgements

MDAnalysis thanks [NumFOCUS][] for its continued support as our fiscal sponsor and 
the [Chan Zuckerberg Initiative][] for supporting MDAnalysis under EOSS4 and EOSS5 awards.

> — @IAlibay and @p-j-smith ([release manager][roles]) on behalf of the [MDAnalysis Team][]

Author: ```suggestion
— @IAlibay and @p-j-smith ([release managers][roles]) on behalf of the [MDAnalysis Team][]
```

This is a minor release of the MDAnalysis library, which means that it
contains enhancements, bug fixes, deprecations, and other
backwards-compatible changes.

## Supported environments

This release supports NumPy 2.0+ and offers backwards compatibility
through to NumPy 1.23.2.

Supported Python versions: **3.10, 3.11, 3.12, 3.13**.

Supported Operating Systems:
  - **Linux** (on [x86-64][])
  - **Windows** (on [x86-64][])
  - **MacOS** (on [x86-64][] and [ARM64][])

## Upgrading to MDAnalysis version 2.9.0

To update with `mamba` (or `conda`)  from the [conda-forge channel][] run

```bash
mamba update -c conda-forge mdanalysis
```

To update from [PyPi][] with `pip` run

```bash
python -m pip install --upgrade MDAnalysis
```

For more help with installation see the [installation instructions in the User Guide][]. 
Make sure you are using a Python version compatible with MDAnalysis 
before upgrading (Python >= 3.10).

## Notable changes

For a full list of changes, bug fixes and deprecations see the [CHANGELOG][].

#### Enhancements:

* Support for Gromacs v2024.4 and v2025 TPR files.
* A new "water" keyword for selecting water residues.

> * Support for **[distopia][] 0.4.0** as an optional distance library

Author: @hmacdope can you look over the distopia part, please?

Author: see also PR https://github.com/MDAnalysis/mdanalysis/pull/4970 for doc update
